r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -0,0 +1,172 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+library(dplyr)
+library(stringr)
+
+test_that("sub and gsub", {
+  df <- tibble(x = c("Foo", "bar"))
+
+  for(fixed in c(TRUE, FALSE)) {
+
+    expect_dplyr_equal(
+      input %>%
+        transmute(x = sub("Foo", "baz", x, fixed = fixed)) %>%
+        collect(),
+      df
+    )
+    expect_dplyr_equal(
+      input %>%
+        transmute(x = sub("^B.+", "baz", x, ignore.case = FALSE, fixed = fixed)) %>%
+        collect(),
+      df
+    )
+    expect_dplyr_equal(
+      input %>%
+        transmute(x = sub("Foo", "baz", x, ignore.case = FALSE, fixed = fixed)) %>%
+        collect(),
+      df
+    )
+
+    # the tests below all use ignore.case = TRUE
+    # but base::sub and base::gsub ignore ignore.case = TRUE with a warning when fixed = TRUE
+    # so we can't use expect_dplyr_equal() for the tests below
```

Comment:
I would split these out of your for loop, include the ones you can as `expect_dplyr_equal`, and make explicit the exceptions.

Reply:
Done in 25b869dffe4496ce8cc253f2a3860e6cce09ec1f